docs/network-helpers/reference.md

```diff
@@ -0,0 +1,268 @@
+await helpers.setStorageAt(address, storageSlot, newValue);
+```
+
+### `getStorageAt(address, index, [block])`
+
+Retrieves the data located at the given address, index, and block number.
+
+Parameters:
+
+- `address`: The address to retrieve storage from.
+- `index`: The position in storage.
+- `block`: The block number, or one of `"latest"`, `"earliest"`, or `"pending"`. Defaults to `"latest"`.
+
+Returns: A string containing the hexadecimal value retrieved.
+
+Example:
+
+```ts
+await helpers.getStorageAt(address, storageSlot);
+```
+
+### `impersonateAccount(address)`
+
+Allows Hardhat Network to sign transactions as the given address.
+
+Parameters:
+
+- `address`: The address to impersonate.
+
+Example:
+
+```ts
+await helpers.impersonateAccount(address);
+```
+
+### `stopImpersonatingAccount(address)`
+
+Stops Hardhat Network from impersonating the given address.
+
+Parameters:
+
+- `address`: The address to stop impersonating.
+
+Example:
+
+```ts
+await helpers.stopImpersonatingAccount(address);
+```
+
+## Time helpers
+
+### `latest()`
+
+Returns the timestamp of the latest block.
+
+Example:
+
+```ts
+await helpers.time.latest();
+```
+
+### `latestBlock()`
+
+Returns the number of the latest block.
+
+Example:
+
+```ts
+await helpers.time.latestBlock();
+```
+
+### `increase(amountInSeconds)`
+
+Mines a new block whose timestamp is `amountInSeconds` after the latest block's timestamp.
+
+Parameters:
+
+- `amountInSeconds`: Number of seconds to increase the next block's timestamp by.
+
+Returns: The timestamp of the mined block.
+
+Example:
+
+```ts
+// advance time by one hour and mine a new block
+await helpers.time.increase(3600);
+```
+
+### `increaseTo(timestamp)`
+
+Mines a new block whose timestamp is `timestamp`.
+
+Parameters:
+
+- `timestamp`: Must be bigger than the latest block's timestamp.
+
+Example:
+
+```ts
+await helpers.time.increaseTo(newTimestamp);
+```
+
+### `setNextBlockTimestamp(timestamp)`
+
+Sets the timestamp of the next block but doesn't mine one.
+
+Parameters:
+
+- `timestamp`: Can be `Date` or Epoch seconds. Must be greater than the latest block's timestamp.
+
+Example:
+
+```ts
+// set the timestamp of the next block but don't mine a new block
+await helpers.time.setNextBlockTimestamp(newTimestamp);
+```
+
+## Snapshots
+
+### `takeSnapshot()`
+
+Takes a snapshot of the state of the blockchain at the current block.
+
+Returns: An object with a `restore` method that can be used to reset the network to this state.
+
+Example:
+
+```ts
+// take a snapshot of the current state of the blockchain
+const snapshot = await helpers.takeSnapshot();
+
+// after doing some changes, you can restore to the state of the snapshot
+await snapshot.restore();
+```
+
+## Fixtures
+
+### `loadFixture()`
+
+Executes the given function and takes a snapshot of the blockchain. Each time `loadFixture` is called again with the same function, the blockchain will be restored to that snapshot instead of executing the function again.
```

Comment:
```suggestion
Executes the given function and takes a snapshot of the blockchain. Upon subsequent calls to `loadFixture` with the same function, rather than executing the function again, the blockchain will be restored to that snapshot.
```

Reply:
420fc1e0589f809dd10350a91999432fe769aea0